### DISCUSSION TITLE
Using a container with a scrollbar is problematic due to the calendar redraw

### DISCUSSION THREAD

**Discussion by giorgiofnool** [easepick/easepick#49] [Q&A] [answered]
Hi!

I'm trying to use easepick to replicate https://www.booking.com calendar.
No problem with the "desktop" version but when I try to replicate the mobile version I found that the calendar is completely redrawn after the selection.
This can lead to a problem because the calendar is put in a container with a scrollbar and the scroll position is reset after the click.

Here's 2 video, one from booking.com and the other from a test version I'm building.

https://user-images.githubusercontent.com/5812801/165342798-dfcc10ec-2f9e-4a13-856f-dae774ace3c0.mov

https://user-images.githubusercontent.com/5812801/165342833-eb24b571-a080-4315-990e-89ee89119cd2.mov

Do you have any tips or suggestion?

Thank you very much!

**Comment by wakirin** [comment, marked as answer]
Hi,
Interesting case. You could try remember state of scroll (`scrollTop`?) on `preselect` event (some global variable).
After this you will need to add:
```js
picker.on('view', (e) => {
   const { view } = e.detail;
   if (view === 'Footer') {
     // restore scroll position
   }
});
```

**Reply by giorgiofnool** [reply]
Yes, it confirms what I was trying to do.
I wasn't sure about the events because it fires lots of them but your suggestion to check the `view` property points me in the right direction.
Thanks!

**Comment by hams2night** [comment]
Hello, 

I would like to do the same thing. A vertical scroll of the calendar on mobile, like on booking.com.

There is no scroll position if you apply your css on the .container, by adding these 3 lines in CSS for example : 

```
@media (max-width:480px){
    .container:not(.inline).show{height:75vh;overflow-y: scroll}
    .calendars .calendar:nth-child(n+2){display:block}
    .calendars .calendar:nth-child(n+2)>.header:not(.no-next-month) .next-button{visibility:hidden}
}

```

You also need to set the number of calendars to 12 for example.

**Comment by dejamba** [comment]
get  scroll position of  main:
```
picker.on('preselect', (e) => {
      scrollTopPosition = e.target.querySelector('main').scrollTop;
```
and  call:
 ```
  if (view === 'Footer') {
      if (typeof scrollTop !== 'undefined'){
          e.target.querySelector('main').scrollTop = scrollTopPosition;
      }
```

**Comment by David-Asin** [comment]
Hi, could you please pass the example code? Thank you very much.
